x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -773,35 +774,53 @@ private static void checkRemoteEnrich(LogicalPlan plan, Set<Failure> failures) {
      * @param failures         failures collection to add to
      */
-    private static void checkNotPresentInDisjunctions(
+    private static void checkFullTextSearchDisjunctions(
         Expression condition,
         java.util.function.Function<FullTextFunction, String> typeNameProvider,
         Set<Failure> failures
     ) {
         condition.forEachUp(Or.class, or -> {
-            checkNotPresentInDisjunctions(or.left(), or, typeNameProvider, failures);
-            checkNotPresentInDisjunctions(or.right(), or, typeNameProvider, failures);
+            boolean left = checkFullTextSearchInDisjunctions(or.left());
+            boolean right = checkFullTextSearchInDisjunctions(or.right());
+            if (left ^ right) {
+                Holder<String> elementName = new Holder<>();
+                if (right) {
+                    or.left().forEachDown(FullTextFunction.class, ftf -> elementName.set(typeNameProvider.apply(ftf)));
+                } else {
+                    or.right().forEachDown(FullTextFunction.class, ftf -> elementName.set(typeNameProvider.apply(ftf)));
+                }
+                failures.add(
+                    fail(
+                        or,
+                        "Invalid condition [{}]. {} can be used as part of an OR condition, "
+                            + "but only if other full text functions are used as part of the condition",
+                        or.sourceText(),
+                        elementName.get()
+                    )
+                );
+            }
         });
     }
 
     /**
-     * Checks whether a condition contains a disjunction with the specified typeToken. Adds to failure if it does.
+     * Checks whether an expression contains just full text functions or negations (NOT) and combinations (AND, OR) of full text functions
      *
-     * @param parentExpression parent expression to add to the failure message
-     * @param or               disjunction that is being checked
-     * @param failures         failures collection to add to
+     * @param expression parent expression to add to the failure message
+     * @return true if all children are full text functions or negations of full text functions, false otherwise
      */
-    private static void checkNotPresentInDisjunctions(
-        Expression parentExpression,
-        Or or,
-        java.util.function.Function<FullTextFunction, String> elementName,
-        Set<Failure> failures
-    ) {
-        parentExpression.forEachDown(FullTextFunction.class, ftp -> {
-            failures.add(
-                fail(or, "Invalid condition [{}]. {} can't be used as part of an or condition", or.sourceText(), elementName.apply(ftp))
-            );
-        });
+    private static boolean checkFullTextSearchInDisjunctions(Expression expression) {
+        if (expression instanceof FullTextFunction) {
+            return false;
```

Comment:
I am having a bit of trouble understanding what this function is doing.
From the javadoc:
`Checks whether an expression contains just full text functions or negations (NOT) and combinations (AND, OR) of full text functions`
but then we return false is the expression is an instance of `FullTextFunction`?
should we rename the function to make it more explicit what it does? this way we remove any potential confusion with the other `checkFullTextSearchDisjunctions` method.